### ISSUE TITLE
--compress-program option for xsv split?

### ISSUE THREAD

**Issue by d33tah** [BurntSushi/xsv#303]
Today I learned how fast `xsv split` can be. The only thing that I feel that it's missing is a `--compress-program` switch (or similar) that would allow the user to compress created split files. @BurntSushi do you feel like it's a good fit for this tool?

**Comment by BurntSushi** [comment]
Nope. It seems totally orthogonal to me. You can always compress them after the split command.

**Comment by d33tah** [comment]
@BurntSushi Thanks for replying! The problem is that you need to store files in the uncompressed form in the meantime. Which, for huge CSV files - which benefit from splitting the most - can be quite cumbersome. In my case I decided to mount a compressed btrfs system to satisfy xsv's requirements, but having a way to compress data before saving to the disk would IMO solve the problem better.

**Comment by LemmingAvalanche** [comment]
@d33tah See [qsv snappy](https://github.com/jqnatividad/qsv/blob/master/src/cmd/snappy.rs#L2).

**Comment by d33tah** [comment]
@LemmingAvalanche thanks!
